pkg/datarepair/checker/checker.go

```diff
@@ -32,71 +33,60 @@ type Config struct {
 	Interval time.Duration `help:"how frequently checker should audit segments" default:"30s"`
 }
 
-// Checker is the interface for data repair checker
-type Checker interface {
-	// TODO: remove interface
-	Run(ctx context.Context) error
-	IdentifyInjuredSegments(ctx context.Context) (err error)
-	OfflineNodes(ctx context.Context, nodeIDs storj.NodeIDList) (offline []int32, err error)
-	Close() error
-}
-
 // Checker contains the information needed to do checks for missing pieces
-type checker struct {
+type Checker struct {
 	statdb      statdb.DB
 	pointerdb   *pointerdb.Service
 	repairQueue queue.RepairQueue
 	overlay     pb.OverlayServer
 	irrdb       irreparable.DB
 	limit       int
 	logger      *zap.Logger
-	ticker      *time.Ticker
+	Interval    sync2.Cycle
```

Comment:
Wondering whether there's a better name for this, like `Identify` or `Check`? But `Interval` is fine as well.